src/librustc_resolve/lib.rs

```diff
@@ -2266,6 +2266,24 @@ impl<'a> Resolver<'a> {
             if !candidates.is_empty() {
                 // Report import candidates as help and proceed searching for labels.
                 show_candidates(&mut err, &candidates, def.is_some());
+            } else {
```

Comment:
Could you add an UI test where the found enum is not in scope? How does `X` in "did you mean to use `X`?" look in this case?